### PR TITLE
[SandboxIR][NFC] Move User into a separate file

### PR DESCRIPTION
**Pull request by vporpo**

### CODE REVIEW

**Comment by vporpo** [comment]
Fixed the clang-format issue.

File: llvm/include/llvm/SandboxIR/User.h [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    return make_range<const_op_iterator>(op_begin(), op_end());
+  }
+  Value *getOperand(unsigned OpIdx) const { return getOperandUse(OpIdx).get(); }
+  /// \Returns the operand edge for \p OpIdx. NOTE: This should also work for
+  /// OpIdx == getNumOperands(), which is used for op_end().
+  Use getOperandUse(unsigned OpIdx) const {
+    return getOperandUseInternal(OpIdx, /*Verify=*/true);
+  }
+  virtual unsigned getNumOperands() const {
+    return isa<llvm::User>(Val) ? cast<llvm::User>(Val)->getNumOperands() : 0;
```
Comment (tschuett) [inline llvm/include/llvm/SandboxIR/User.h:129]: dyn_cast!

Comment (vporpo) [reply llvm/include/llvm/SandboxIR/User.h:129]: Let's do these types of changes separately. For these NFC patches I will be moving code around without modifying it.

Comment (tschuett) [reply llvm/include/llvm/SandboxIR/User.h:129]: It still NFC.

Comment (vporpo) [reply llvm/include/llvm/SandboxIR/User.h:129]: Yes, but if for whatever reason I mess up the dyn_cast change I would have to revert the whole code-moving patch which is not ideal. It is safer to change one thing at a time.

**Review by Sterling-Augustine — APPROVED** [review]

**Pull request merged**